### PR TITLE
ST-1758: Add empty view component

### PR DESCRIPTION
**Pull request by ilyawzrd**
Description:

--

QA-test: ST-1760

Demo-test: ST-1762

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/client-app-0.0.4-pr-133-db6914f3.zip

### CODE REVIEW

File: client-app/pages/account/addresses.vue [hunk 1]
```diff
@@ -14,13 +17,35 @@
...
           </div>
 
-          <div class="flex flex-col bg-white shadow-sm md:rounded md:border">
+          <!-- Empty view -->
+          <VcEmptyView
+            :text="$t('pages.account.addresses.no_addresses_message')"
+            v-if="!paginatedAddresses.length && !editingMode"
+          >
+            <template #icon>
+              <VcImage :src="'/static/images/account/icons/no-addresses.svg'" />
```
Comment (Flanker72) [inline client-app/pages/account/addresses.vue:38]: Use `src="/static/images/account/icons/no-addresses.svg"` instead - you shouldn't pass simple string like variable.

File: client-app/components/atoms/empty-view/empty-view.vue [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<template>
+  <div class="flex grow flex-col items-center justify-center space-y-5" :class="isMobile && 'h-96'">
+    <slot name="icon"></slot>
+    <p class="text-xl">
```
[the remark this answers is not in the document]
Comment (ilyawzrd) [reply client-app/components/atoms/empty-view/empty-view.vue]: done

File: client-app/components/atoms/empty-view/empty-view.vue [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      {{ text }}
+    </p>
+    <slot name="button"></slot>
+  </div>
+</template>
+<script setup lang="ts">
```
[the remark this answers is not in the document]
Comment (ilyawzrd) [reply client-app/components/atoms/empty-view/empty-view.vue:12]: done

File: client-app/pages/account/list-details.vue [hunk 4]
```diff
@@ -77,15 +77,16 @@
           </template>
 
           <!-- Empty -->
-          <div v-else class="flex grow items-center justify-center">
-            <div class="text-center my-20 lg:my-0 p-6">
-              <p class="font-bold mb-7 text-xl md:text-2xl">{{ $t("shared.wishlists.list_details.empty_list") }}</p>
-
+          <VcEmptyView :text="$t('shared.wishlists.list_details.empty_list')" v-else>
+            <template #icon>
+              <VcImage src="/static/images/common/list.svg" :alt="list?.name" />
```
[the remark this answers is not in the document]
Comment (ilyawzrd) [reply client-app/pages/account/list-details.vue]: done

File: client-app/pages/account/list-details.vue [hunk 5]
```diff
@@ -77,15 +77,16 @@
           </template>
 
           <!-- Empty -->
-          <div v-else class="flex grow items-center justify-center">
-            <div class="text-center my-20 lg:my-0 p-6">
-              <p class="font-bold mb-7 text-xl md:text-2xl">{{ $t("shared.wishlists.list_details.empty_list") }}</p>
-
+          <VcEmptyView :text="$t('shared.wishlists.list_details.empty_list')" v-else>
+            <template #icon>
+              <VcImage :src="'/static/images/common/list.svg'" :alt="list?.name" />
```
[the remark this answers is not in the document]
Comment (ilyawzrd) [reply client-app/pages/account/list-details.vue]: done

**Pull request merged**